### PR TITLE
Only depend on transformers-compat on old GHCs

### PR DESCRIPTION
**Pull request by RyanGlScott**
See https://gitlab.haskell.org/ghc/ghc/issues/16752.

### CODE REVIEW

File: exceptions.cabal [hunk 1]
```diff
@@ -53,6 +57,14 @@ library
   if !impl(ghc >= 8.0)
```
Comment (bgamari) [inline exceptions.cabal]: Don't you need to remove the `transformers-compat` dependency [above](https://github.com/ekmett/exceptions/pull/74/files#diff-451f08946ce59d8d252355dc3fc83332R54)?

Comment (RyanGlScott) [reply exceptions.cabal]: Oops. Yes, good catch.

**Comment by bgamari** [comment]
I'll give this a shot in the GHC build.

**Comment by bgamari** [comment]
Indeed this works as advertised. Thanks @RyanGlScott!

**Pull request merged**